ratis-server/src/main/java/org/apache/ratis/server/impl/LeaderStateImpl.java

```diff
@@ -1110,8 +1109,9 @@ CompletableFuture<Long> getReadIndex() {
   }
 
   @Override
-  public void onAppendEntriesReply(FollowerInfo follower, RaftProtos.AppendEntriesReplyProto reply) {
-    server.getState().getReadRequests().onAppendEntriesReply(reply, this::hasMajority);
+  public void onAppendEntriesReply(LogAppender appender, FollowerInfo follower,
+                                   RaftProtos.AppendEntriesReplyProto reply) {
+    server.getState().getReadRequests().onAppendEntriesReply(appender, reply, this::hasMajority);
```

Comment:
One more comment: Use `server.getReadRequests()`  instead in line 1114 (this line) and line 1090.

Reply:
done